pandas/tests/frame/test_to_csv.py

```diff
@@ -1143,3 +1143,34 @@ def test_to_csv_quoting(self):
         df = df.set_index(['a', 'b'])
         expected = '"a","b","c"\n"1","3","5"\n"2","4","6"\n'
         self.assertEqual(df.to_csv(quoting=csv.QUOTE_ALL), expected)
+
+    def test_period_index_date_overflow(self):
+        # see gh-15982
+
+        dates = ["1990-01-01", "2000-01-01", "3005-01-01"]
+        index = pd.PeriodIndex(dates, freq="D")
+        df = pd.DataFrame([4, 5, 6], index=index)
+
+        buf = StringIO()
```

Comment:
Actually, now that I look at this, since `df.to_csv` DOES return if `buf` is None, complicating the testing with writing to a buffer seems unnecessary.  In the future, we should encourage any tests of `to_csv` to do the following:
~~~python
expected = "foo"
result = df.to_csv()
assert result == expected
~~~